api/converter_utils.go

```diff
@@ -34,8 +34,18 @@ func decodeDigest(str *string, field string, errorArr []string) (string, []strin
 	return "", errorArr
 }
 
-// decodeAddress returns the byte representation of the input string, or appends an error to errorArr
-func decodeAddress(str *string, field string, errorArr []string) ([]byte, []string) {
+// decodeAddress returns the sdk.Address representation of the input string, or appends an error to errorArr
+func decodeAddress(str string, field string, errorArr []string) (sdk.Address, []string) {
```

Comment:
I see that `decodeAddress` already existed, so this PR is not to blame for the unusual way that multiple errors are accumulated, but I think there is a better way to handle multiple errors these days.

I would use `sdk.DecodeAddress` directly, and if it returns an `error`, append that into a `[]error` instead of `[]string`.  Use `fmt.Errorf` with a `%w` to produce an error that wraps the error from `sdk.DecodeAddress` but also contains the `str` that would have been passed into `decodeAddress`), rather than into a slice of `string`.  Then at the end of a function that may have accumulated errors, test its size just as you do, but then use `errors.Join(errArr...)` to combine them into a single `error`.

I'll leave it to others to decide if this is important enough to change. I believe it will make it easier to follow the error handling, but it isn't critical if this is a common pattern in `indexer`.

Reply:
Oh, even nicer: 
```Join returns an error that wraps the given errors. Any nil error values are discarded. Join returns nil if every value in errs is nil.```
therefore, you don't need to test the `err` that is returned from `sdk.DecodeAddress`, just append it to your error list.  Then, at the bottom, don't check to see if the list has errors in it, just:
```
  err := errors.Join(errList...)
  if err != nil {
    return err
  }
```

Reply:
See https://github.com/algorand/indexer/pull/1638#discussion_r1853159861